url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/ReadConfig.java

```diff
@@ -0,0 +1,226 @@
+package ccproxy;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class ReadConfig {
+	boolean configOk = false;
+	String pghost = null;
+	String pgport = "5432";
+	String pguser = null;
+	String pgpass = null;
+	String pgdbname = null;
+
+	String asteriskHost = null;
+	String asteriskPort = "5038";
+	String asteriskUser = null;
+	String asteriskSecret = null;
+	String listenPort = null;
+
+	String licensedAgents=null;
+	String licenseKey=null;
+	String licenseExpires="";
+	
+	String agentVersion = null;
+	String agentUpdateURL = null;
+	String agentContext = null;
+	// Czy po zakończeniu połączenia telefonicznego agent powienien przejść w tryb pauzy
+	boolean pauseAfterConnection = false;
+	// Czy po zalogowaniu agent powinien pozostać w trybie pauza
+	boolean pauseWhenStartup = false;
+	// Czy logować eventy płynące z asteriska.
+	boolean logAsteriskEvents = false;
+	String externAuthURL = null;
+	boolean monitorDbActivity = false;
+	boolean pauseWhenUnavailable = false;
+	boolean requestCallTag = false;
+	String configFile;
+
+	
+	String spyMode="";
+	
+	org.apache.log4j.Logger log;
+	
+
+	public ReadConfig(Parametry p, String filename) {
+
+		log = p.log;
+		FileReader Fr;
+		try {
+			configFile=filename;
+			p.log.info("Trying to open config file [" + filename + "]");
+			Fr = new FileReader(filename);
+			BufferedReader br = new BufferedReader(Fr);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. Currently, there are execution paths that do not contain closure statements, for example, when an else condition at lines 198, 203, 208, 193 evaluate to true. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).